kotlin-eclipse-gradle/scripts/init.gradle.kts

```diff
@@ -0,0 +1,56 @@
+import javax.inject.Inject
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.tooling.provider.model.ToolingModelBuilder
+import org.gradle.tooling.provider.model.ToolingModelBuilderRegistry
+import org.jetbrains.kotlin.gradle.model.GradleProjectForEclipse
+import org.jetbrains.kotlin.gradle.model.GradleProjectForEclipseImpl
+import org.jetbrains.kotlin.gradle.model.NoKotlinProject
+
+initscript {
+    dependencies {
+        classpath(files(System.getProperty("org.jetbrains.kotlin.eclipse.gradle.model.path")))
+    }
+}
+
+allprojects {
+    buildscript {
+        dependencies {
+            classpath(files(System.getProperty("org.jetbrains.kotlin.eclipse.gradle.model.path")))
+        }
+    }
+
+    apply<GradleProjectForEclipseInstaller>()
+}
+
+class GradleProjectForEclipseInstaller @Inject constructor(val registry: ToolingModelBuilderRegistry) : Plugin<Project> {
+    override fun apply(project: Project) {
+        registry.register(GradleProjectForEclipseBuilder())
+    }
+}
+
+class GradleProjectForEclipseBuilder() : ToolingModelBuilder {
+    override fun canBuild(modelName: String) = modelName == GradleProjectForEclipse::class.qualifiedName
+
+    override fun buildAll(modelName: String, project: Project): Any {
```

Comment:
Minor. The return type is `Any`? 
JFYI: Return types can be more specific that they are in the superclass.